AFMReader/h5_jpk.py

```diff
@@ -0,0 +1,349 @@
+"""
+Module to decode and load .h5-jpk AFM file format into 3D Python NumPy arrays.
+
+It extracts scan channels, reshapes image frames, applies scaling, and generates
+timestamps based on scan metadata.
+"""
+
+from pathlib import Path
+
+import h5py
+import numpy as np
+
+from AFMReader.logging import logger
+
+logger.enable(__package__)
+
+
+def _parse_channel_name(channel: str) -> tuple[str, str]:
+    """
+    Validate and split a channel name into its type and trace direction.
+
+    Parameters
+    ----------
+    channel : str
+        The name of the channel, expected in the form 'name_trace' or 'name_retrace'.
+
+    Returns
+    -------
+    tuple[str, str]
+        A tuple containing the channel type and trace type.
+
+    Raises
+    ------
+    ValueError
+        If the format is invalid or the trace type is not 'trace' or 'retrace'.
+    """
+    if "_" not in channel:
+        raise ValueError(f"Invalid channel format '{channel}'. Expected 'name_trace' or 'name_retrace'.")
+
+    channel_type, trace_type = channel.rsplit("_", 1)
+    trace_type = trace_type.lower()
+
+    if trace_type not in ("trace", "retrace"):
+        raise ValueError(f"Invalid trace type '{trace_type}'. Must be 'trace' or 'retrace'.")
+
+    return channel_type, trace_type
+
+
+def _get_channel_info(f: h5py.File, channel: str):
```

Comment:
Not a great fan of single character parameter names as they can be confusing (perhaps just me though) and prefer descriptive names, suggest renaming to `h5py_file` throughout the function.

```suggestion
def _get_channel_info(h5py_file: h5py.File, channel: str):
```